src/main/java/at/hannibal2/skyhanni/data/BitsAPI.kt

```diff
@@ -0,0 +1,141 @@
+        "§eYou gained §3(?<amount>.*) Bits Available §ecompounded from all your §epreviously eaten §6cookies§e! Click here to open §6cookie menu§e!"
+    )
+    private val bitsEarnedChatPattern by group.pattern("chat.earned", "§f\\s+§8\\+§b(?<amount>.*)\\s+Bits\n")
+    private val boosterCookieAte by group.pattern("chat.boostercookie.ate", "§eYou consumed a §6Booster Cookie§e!.*")
+    private val bitsAvailableMenu by group.pattern(
+        "gui.bitsavailablemenu",
+        "§7Bits Available: §b(?<toClaim>[\\w,]+)(§3.+)?"
+    )
+    private val fameRankSbmenu by group.pattern("gui.fameranksbmenu", "§7Your rank: §e(?<rank>.*)")
+    private val fameRankCommunityShop by group.pattern("gui.famerankcommunityshop", "§7Fame Rank: §e(?<rank>.*)")
+    private val bitsGuiNamePattern by group.pattern("gui.mainmenuname", "^SkyBlock Menu$")
+    private val bitsGuiStackPattern by group.pattern("gui.mainmenustack", "^§6Booster Cookie$")
+    private val fameRankGuiNamePattern by group.pattern("gui.famerankmenuname", "^(Community Shop|Booster Cookie)$")
+    private val fameRankGuiStackPattern by group.pattern("gui.famerankmenustack", "^(§aCommunity Shop|§eFame Rank)$")
+
+    @SubscribeEvent
+    fun onScoreboardChange(event: ScoreboardChangeEvent) {
+        for (line in event.newList) {
+            val message = line.trimWhiteSpaceAndResets()
+
+            bitsScoreboardPattern.matchMatcher(message) {
+                val amount = group("amount").formatNumber().toInt()
+                val earned = group("earned")?.formatNumber()?.toInt() ?: 0
+                bits = amount
+                if (earned > 0) bitsToClaim -= earned
+
+                return
+            }
+        }
+    }
+
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) {
+        val message = event.message.trimWhiteSpaceAndResets()
+
+        bitsFromFameRankUpChatPattern.matchMatcher(message) {
+            val amount = group("amount").formatNumber().toInt()
+            bitsToClaim += amount
+
+            return
+        }
+
+        bitsEarnedChatPattern.matchMatcher(message) {
+            // Only two locations where the bits line isn't shown, but you can still get bits
+            if (LorenzUtils.inAnyIsland(IslandType.CATACOMBS, IslandType.THE_RIFT)) return
+
+            val amount = group("amount").formatNumber().toInt()
+            bits += amount
+            bitsToClaim -= amount
+
+            return
+        }
+
+        boosterCookieAte.matchMatcher(message) {
+            bitsToClaim += (defaultcookiebits * currentFameRank.bitsMultiplier).toInt()
+
+            return
+        }
+    }
+
+    @SubscribeEvent
+    fun onInventoryFullyLoaded(event: InventoryFullyOpenedEvent) {
+        if (!LorenzUtils.inSkyBlock) return
+
+        val stacks = event.inventoryItems
+
+        if (bitsGuiNamePattern.matches(event.inventoryName)) {
+            val cookieStack = stacks.values.firstOrNull { bitsGuiStackPattern.matches(it.displayName) }
```

Comment:
Same code but slightly faster
```suggestion
            val cookieStack = stacks.values.lastOrNull { bitsGuiStackPattern.matches(it.displayName) }
```